src/DynamoCore/Core/UndoRedoRecorder.cs

```diff
@@ -398,8 +398,12 @@ private void UndoActionGroup(XmlElement actionGroup)
                         try
                         {
                             ModelBase toBeDeleted = undoClient.GetModelForElement(element);
-                            RecordActionInternal(newGroup, toBeDeleted, modelActionType);
-                            undoClient.DeleteModel(element);
+                            if (toBeDeleted != null)
```

Comment:
I guess after looking at the code, I still failed to understand why this check is a must here

Reply:
for this case I guess it is not required since we are inside the try block, but note that in many cases the catch block below does rethrow the exception - but I agree, I do not want to change behavior here because I am not familiar with `isOffTrack` - for this case I will get rid of the null check and just let it ride! 🐴

Reply:
second thoughts.